pull/github.go

```diff
@@ -927,6 +976,86 @@ func (ga v4GitActor) GetV3Login() string {
 	return ""
 }
 
+type v4PermissionSource struct {
+	Type       string `graphql:"__typename"`
+	Permission string
+}
+
+type v4PermissionSources []v4PermissionSource
+
+func (pss v4PermissionSources) IsViaOrg(combined Permission) bool {
```

Comment:
I'm not a fan of this function; will probably revisit tomorrow and might switch it to make a few more API requests to avoid assuming quite so much about how the API will behave in the future.

At minimum, it needs a full set of tests for the described functionality.

Reply:
Replaced in latest update. I think the extra API requests (which should only be 1 in practice) make it easier to understand and less fragile to future API changes.